app/helpers/export_helper.rb

```diff
@@ -341,6 +361,30 @@ def extract_patients_details(patients, fields)
             patient_details[:vaccine_2_dose_number] = patients_vaccines[patient.id].second[:dose_number] if fields.include?(:vaccine_2_dose_number)
             patient_details[:vaccine_2_notes] = patients_vaccines[patient.id].second[:notes] if fields.include?(:vaccine_2_notes)
           end
+
+          if patients_vaccines[patient.id]&.third&.present?
+            patient_details[:vaccine_3_group_name] = patients_vaccines[patient.id].third[:group_name] if fields.include?(:vaccine_3_group_name)
+            patient_details[:vaccine_3_product_name] = patients_vaccines[patient.id].third[:product_name] if fields.include?(:vaccine_3_product_name)
+            if fields.include?(:vaccine_3_administration_date)
+              patient_details[:vaccine_3_administration_date] = patients_vaccines[patient.id].third[:administration_date]&.strftime('%F')
+            end
+            patient_details[:vaccine_3_dose_number] = patients_vaccines[patient.id].third[:dose_number] if fields.include?(:vaccine_3_dose_number)
+            patient_details[:vaccine_3_notes] = patients_vaccines[patient.id].third[:notes] if fields.include?(:vaccine_3_notes)
+          end
+        end
+
+        # populate common export cohorts if requested (export for all monitorees)
+        if patients_cohorts&.key?(patient.id)
+          if patients_cohorts[patient.id]&.first&.present?
+            patient_details[:cohort_1_type] = patients_cohorts[patient.id].first[:cohort_type] if fields.include?(:cohort_1_type)
+            patient_details[:cohort_1_name] = patients_cohorts[patient.id].first[:cohort_name] if fields.include?(:cohort_1_name)
+            patient_details[:cohort_1_location] = patients_cohorts[patient.id].first[:cohort_location] if fields.include?(:cohort_1_location)
+          end
+          if patients_cohorts[patient.id]&.second&.present?
+            patient_details[:cohort_2_type] = patients_cohorts[patient.id].first[:cohort_type] if fields.include?(:cohort_2_type)
+            patient_details[:cohort_2_name] = patients_cohorts[patient.id].first[:cohort_name] if fields.include?(:cohort_2_name)
+            patient_details[:cohort_2_location] = patients_cohorts[patient.id].first[:cohort_location] if fields.include?(:cohort_2_location)
```

Comment:
These are still using the `first` cohort. Should be `.second`

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1125/commits/11f9765d3d0d6b687f0a3e8f6d62ab6192eadf4b#diff-3346374057035485f7d2ab03927877c081ce3af8eff7a01994fe76211256d9ebR384